src/http/client.cc

```diff
@@ -19,10 +19,79 @@
  * Copyright (C) 2022 Scylladb, Ltd.
  */
 
+#include <seastar/core/loop.hh>
+#include <seastar/core/when_all.hh>
 #include <seastar/http/client.hh>
+#include <seastar/http/request.hh>
+#include <seastar/http/reply.hh>
+#include <seastar/http/response_parser.hh>
 
 namespace seastar {
 namespace http {
 
+client::client(connected_socket&& fd)
+        : _fd(std::move(fd))
+        , _read_buf(_fd.input())
+        , _write_buf(_fd.output())
+{
+}
+
+future<> client::write_body(request& req) {
+    if (req.body_writer) {
+        return req.body_writer(_write_buf);
+    } else {
+        return make_ready_future<>();
+    }
+}
+
+future<> client::send_request(request& req) {
+    if (req._version.empty()) {
+        req._version = "1.1";
```

Comment:
There's quite a bit of details missing headers in this implementation, such as host (super-important!), connection:close, date, etc., that should be in HTTP 1.1.
One can argue that the user should specify these headers themselves with the rest of the headers. But isn't that very inconvenient? APIs from Python and other languages don't force you to do that.
We can do this later, but need to think about leaving FIXMEs, etc., or nobody will remember what still needs fixing.

Reply:
"Host:" -- yes, should be there. As for other headers -- I didn't find any unconditional "MUST"s about them in the RFC, so will leave FIXMEs around in the next version.